app/models/my_module_repository_row.rb

```diff
@@ -13,17 +16,28 @@ class MyModuleRepositoryRow < ApplicationRecord
 
   around_save :deduct_stock_balance, if: :stock_consumption_changed?
 
+  before_save :nulify_stock_consumption, if: :stock_consumption_changed?
+
   private
 
+  def nulify_stock_consumption
+    self.stock_consumption = nil if stock_consumption.zero?
+  end
+
   def deduct_stock_balance
     stock_value = repository_row.repository_stock_value
-    delta = stock_consumption_was.to_d - stock_consumption.to_d
-    lock!
```

Comment:
why it is removed?

Reply:
You can't lock a changed object, so I moved it out before the row changed.

Reply:
`RuntimeError: Locking a record with unpersisted changes is not supported. Use 'save' to persist the changes, or 'reload' to discard them explicitly.`